### PR TITLE
Bug 1840090 - Add content strings for shopping experience

### PR DESCRIPTION
**Pull request by Alexandru2909**
Added updated strings for review quality check feature.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1840090

### CODE REVIEW

File: fenix/app/src/main/res/values/static_strings.xml
Comment (calumozilla) [inline fenix/app/src/main/res/values/static_strings.xml:30]: Maybe `shopping_product_name` to avoid using the word Fakespot?

**Review by calumozilla — APPROVED** [review]

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2081]: I don't think we've got the r+ from branding on "Fakespot by Mozilla" yet (Slack [discussion](https://mozilla.slack.com/archives/C0568AAEX2A/p1692384186860099)), so we shouldn't be landing this string until then

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2083]: Again let's explain this grading system.
Also, why is "letter grade" not just part of this string and instead replaced by a parameter? 🤔

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2086]: Again explain grading system

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2088]: explain grading system

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2090]: explain grading system

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2093]: Again, I'm not sure why we are using a parameter here instead of just one string?

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2097]: Same question about parameter vs just having one string

**Review by Delphine — CHANGES_REQUESTED** [review]
Left some comments but I'm stopping my review at my last comment before continuing.

What I'm noticing is that the comments in this PR sound more like developer comments vs [localization notes](https://mozilla-l10n.github.io/documentation/localization/dev_best_practices.html#add-localization-notes) - they're often not constructed as full sentences so it can be confusing to understand them without context. I'm also assuming all the mentions of "cards" in comments refer to FIGMA cards? If so, it would be best to take them out as it will create confusion for translation.
Please keep in mind localizers rarely have context other than the string and its comment, and that they need to have some details in order to translate correctly (they also often only have visibility over one string at a time). I would suggest to clean up comments to give more context and be constructed sentences.

Also, I might be missing something, why are so many strings cut in two strings with a parameter referring to another string? 🤔 

Anyways, wanted to call this out before continuing this lengthy review. Thank you.

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [reply fenix/app/src/main/res/values/strings.xml:2064]: Update: we can actually communicate this directly with localizers, so going to resolve this comment and similar ones.

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [reply fenix/app/src/main/res/values/strings.xml:2072]: An example/suggestion for an l10n comment in this case would be: "Title/headline for when the rating has been updated by the review checker"
I think you can generally use this example/style and adapt to other comments for strings in the PR

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [reply fenix/app/src/main/res/values/strings.xml:2081]: Update: I see we now did get a response since yesterday and the "by" will be localized

**Review by Delphine — COMMENTED** [review]
I've updated a few things since my last comment, and some more:

* l10n team can tell localizers about the grading system, no need to update that in comments
* I gave an example of an l10n note and how to update similarly across comments in PR
* Is it possible to use markup to create whole sentences vs. splitting strings using variables? This helps avoid getting sometimes weird translation results, since localizers only see one string at a time 😄

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2200]: Please explain placeholder in comment

File: fenix/app/src/main/res/values/strings.xml
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2202]: Seems like the remaining comments in strings could add more context, in the same way that all the other comments were updated

**Review by Delphine — CHANGES_REQUESTED** [review]
Thanks for the update! Comments look much better now, and I've just left a note at the end because the last strings look like they could have similar comments and give more context.

Quick note: (this was already raised with UX but should be ok from the feedback I got) strings are longer than usual and localized strings are at least 25% longer than English (such as French and German), so hopefully this will look good on a smaller screen such as mobile

I'll wait for a response about avoiding splitting the strings. Indeed, after doing a full pass today, I'm really afraid the use of so many placeholders for so many sentences / instances will lead to wrong sentences once they are patched up. I can't find an example of when we've done this in the past, but maybe another Android dev could share some insight? thank you

**Comment by flodolo** [comment]
> I'll wait for a response about avoiding splitting the strings. Indeed, after doing a full pass today, I'm really afraid the use of so many placeholders for so many sentences / instances will lead to wrong sentences once they are patched up. I can't find an example of when we've done this in the past, but maybe another Android dev could share some insight? thank you

Not sure if this helpful, but we have strings using markup through CDATA, at lest for formatting. Here's a couple:

```
    <string name="pair_instructions_2"><![CDATA[Scan the QR code shown at <b>firefox.com/pair</b>]]></string>
    <string name="sign_in_create_account_text"><![CDATA[No account? <u>Create one</u> to sync Firefox between devices.]]></string>
```

Variable replacement is normally used only for brand names, or dynamic content (e.g. URL of the website). As Delphine mentioned, this is likely to lead to low quality translations, because it's incredibly hard to translate sentences correctly, when content is split in different strings.

**Comment by Alexandru2909** [comment]
> > I'll wait for a response about avoiding splitting the strings. Indeed, after doing a full pass today, I'm really afraid the use of so many placeholders for so many sentences / instances will lead to wrong sentences once they are patched up. I can't find an example of when we've done this in the past, but maybe another Android dev could share some insight? thank you
> 
> Not sure if this helpful, but we have strings using markup through CDATA, at lest for formatting. Here's a couple:
> 
> ```
>     <string name="pair_instructions_2"><![CDATA[Scan the QR code shown at <b>firefox.com/pair</b>]]></string>
>     <string name="sign_in_create_account_text"><![CDATA[No account? <u>Create one</u> to sync Firefox between devices.]]></string>
> ```
> 
> Variable replacement is normally used only for brand names, or dynamic content (e.g. URL of the website). As Delphine mentioned, this is likely to lead to low quality translations, because it's incredibly hard to translate sentences correctly, when content is split in different strings.

This is a known limitation of the toolkit we're currently using for UI (Compose) as it doesn't support CDATA formatting. This means we'll have two options to avoid the risk of low quality translations:
 - We can use an `AndroidView` and display the spanned text in a `TextView` as detailed [here](https://developer.android.com/codelabs/jetpack-compose-migration?continue=https%3A%2F%2Fdeveloper.android.com%2Fcourses%2Fpathways%2Fcompose%23codelab-https%3A%2F%2Fdeveloper.android.com%2Fcodelabs%2Fjetpack-compose-migration#8).
 - We can write our own HTML to `SpannedString` parser, as was introduced in the `parseHtml` method from this PR.
 
I'll update the highlighted strings using CDATA for now and wait for feedback from the dev team to choose the approach.

File: fenix/app/src/main/res/values/strings.xml
Comment (Alexandru2909) [reply fenix/app/src/main/res/values/strings.xml]: Just to be sure, I also corrected `review_quality_check_explanation_body_adjusted_grading`, "The adjusted rating is <s>a</s> based only..".

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2105]: Are we limited to text-level tags (like bold) for CDATA? Could we use the same approach for links or that's out of the question?

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2110]:
```suggestion
    <!-- Text for label for switch preference to show recommended products from review quality check settings section. -->
```

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2112]:
```suggestion
    <!-- Description for switch preference to show recommended products from review quality check settings section. First parameter is for clickable text defined in review_quality_check_settings_recommended_products_learn_more.-->
```

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2175]: Firefox should be a parameter as well here.

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2174]: The comment is incomplete, as there's at least one parameter for Fakespot, and there should be one for Firefox.

**Review by flodolo — CHANGES_REQUESTED** [review]

File: fenix/app/src/main/res/values/strings.xml
Comment (Alexandru2909) [reply fenix/app/src/main/res/values/strings.xml]: I don't see the word Fakespot being used in this string. Is this referring to another string?

File: fenix/app/src/main/res/values/strings.xml
Comment (Alexandru2909) [reply fenix/app/src/main/res/values/strings.xml:2105]: The same approach can be used for links, I'll update the method in this PR to support links and start a conversation with the dev team.

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [reply fenix/app/src/main/res/values/strings.xml]: I misread the string, I thought it was referring to Fakespot, but it's one of the 3 ecommerce sites.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt:349]: Now this is removed, we can remove the `Suppress("Depecation")` annotation

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [reply fenix/app/src/main/res/values/strings.xml:2105]: Any update on the outcome of the conversation?

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt
Comment (Alexandru2909) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt:349]: The `ClickableSubstringLink` was actually moved lower to match the updated design. We'll still need to suppress deprecation for that. I'll keep the current position and let the design inconsistencies to be fixed in 1848219.

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2200]:
```suggestion
    <!-- Description for review quality check contextual onboarding card. The first and last two parameters are for retailer names (e.g. Amazon, Walmart). The second parameter is for the name of the application (e.g. Firefox). -->
```

I don't think Fenix would ever be used as the application name?

**Review by flodolo — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt:349]: Sounds fair, thanks for the clarification!

**Review by rahulsainani — APPROVED** [review]

**Comment by DonalMe** [comment]
@Alexandru2909 this has pending reviewers, mergify expects `pending reviewers = 0` before adding to the merge queue and landing.
Please take a look and clear as needed

**Comment by Alexandru2909** [comment]
@Delphine requested the changes before she went on PTO but I confirmed with @flodolo that they will cover the review for her while she's out. I'm going to clear Delphine's review.

File: fenix/app/src/main/res/values/strings.xml
Comment (flodolo) [inline fenix/app/src/main/res/values/strings.xml:2238]: Spotted while translating: there is a "review checker", not a "Review quality check"? Also, the capitalization is inconsistent with the next string.

**Pull request merged**
